zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -137,15 +163,76 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         );
 
         // Transfer value from caller to callee
+        // Use cb.query_word as TransferWithGasFeeGadget
+        // expects words instead of expressions for tx_value and gas_fee
+        let effective_tx_value = cb.query_word();
+        let effective_gas_fee = cb.query_word();
+        cb.condition(is_tx_invalid.expr(), |cb| {
+            cb.require_equal(
+                "effective_tx_value == 0",
+                effective_tx_value.expr(),
+                true.expr(),
+            );
+            cb.require_equal(
+                "effective_gas_fee == 0",
+                effective_gas_fee.expr(),
+                true.expr(),
+            );
+        });
+
```

Comment:
Very much a nitpick but maybe remove this newline here so the if/else relation is more clear.